src/v/pandaproxy/schema_registry/protobuf.cc

```diff
@@ -168,14 +171,43 @@ ss::future<const pb::FileDescriptor*> import_schema(
   sharded_store& store,
   const canonical_schema& schema) {
     try {
-        parser p;
-        auto fdp = p.parse(schema);
-        co_return build_file(dp, fdp);
+        co_return co_await build_file_with_deps(dp, store, schema);
     } catch (const exception& e) {
         throw as_exception(invalid_schema(schema));
     }
 }
 
+///\brief Build a FileDescriptor and import dependencies from the store.
+///
+/// Recursively import dependencies into the DescriptorPool, building the files
+/// on stack unwind.
+ss::future<const pb::FileDescriptor*> build_file_with_deps(
+  pb::DescriptorPool& dp,
+  sharded_store& store,
+  const canonical_schema& schema) {
+    parser p;
+    auto fdp = p.parse(schema);
+
+    const auto dependency_size = fdp.dependency_size();
+    for (int i = 0; i < dependency_size; ++i) {
+        auto sub = subject{fdp.dependency(i)};
+        if (auto fd = dp.FindFileByName(sub()); !fd) {
```

Comment:
is this doing any sort of filesystem operations?

Reply:
> is this doing any sort of filesystem operations?

No, the textual protobuf definitions are called "files".

Reply:
got it. thanks